olcExampleProgram.cpp

```diff
@@ -60,16 +67,23 @@ class Game : public olc::PixelGameEngine
 
 		// Update NPC positions and render
 		SetPixelMode(olc::Pixel::ALPHA);
-		for (Entity* e : entities) {
+		for (auto& e : entities) {
 
 			// Get the entity's position
 			olc::vf2d pos = e->getPos();
 
+			// Dont render the entity if they are outside the screen boundaries
+			if ((pos + cameraOffsets).x + e->r < 0
+				|| (pos + cameraOffsets).x - e->r > ScreenWidth()
+				|| (pos + cameraOffsets).y + e->r < 0
+				|| (pos + cameraOffsets).y - e->r > ScreenHeight())
+				continue;
+
```

Comment:
Minor optimization... Do not render, update position, or calculate collision for entities off-screen.